File: algorithms/math/josephus-problem/cpp/josephus_problem.cpp
int josephus(int n, int k) {
    if (n <= 0 || k <= 0) {
        return -1;
    }

    int survivor = 0;
    for (int size = 2; size <= n; ++size) {
        survivor = (survivor + k) % size;
    }
    return survivor;
}

File: algorithms/math/josephus-problem/go/josephus_problem.go
package josephusproblem

func josephus(n, k int) int {
	if n <= 0 || k <= 0 {
		return 0
	}

	survivor := 0
	for size := 2; size <= n; size++ {
		survivor = (survivor + k) % size
	}

	return survivor
}
